ethers-providers/src/transports/quorum.rs

```diff
@@ -179,6 +179,36 @@ impl<T: JsonRpcClientWrapper> QuorumProvider<T> {
             .ok_or_else(|| ProviderError::CustomError("No Providers".to_string()))
     }
 
+    /// Returns the block height that a _quorum_ of providers have reached.
+    async fn get_quorum_block_number(&self) -> Result<U64, ProviderError> {
+        // To be even more robust, we could use an iterator and handle as futures respond to be
+        // tolerant to a provider that has stopped responding.
+        let mut numbers = join_all(self.providers.iter().map(|provider| async move {
```

Comment:
iiuc `join_all` will error immediately if any of the requests have an error right? I think we'd definitely want something robust to errors, and ideally like you mention one that is more resilient to some providers taking a while.
Fwiw I'm not _super_ concerned about providers taking a long time affecting things too much because we'll have each provider using a 60s timeout. Related though - I thought this was the case but we'll need to ensure that the `http_provider`s created [here](https://github.com/abacus-network/abacus-monorepo/blob/main/rust/chains/abacus-ethereum/src/trait_builder.rs#L40) have a timeout just like they do [here](https://github.com/abacus-network/abacus-monorepo/blob/main/rust/chains/abacus-ethereum/src/trait_builder.rs#L50) now.

But anyways I assume the easiest way to get resilience to errors is to use something like `select_all`, in which case we should easily be able to handle as futures respond like you mention?

Reply:
`join_all` is not actual the issue here, you will notice that `join_all` returns a list of `T` where it takes in `IntoIterator<Item=Future<Output=T>>` and in this case `T` is a `Result<U64, ProviderError>` so we are good as long as we don't restructure with the collect and then `?` it away. 

See the new version I wrote which partitions and then returns a quorum error if it fails to reach a quorum.